scipy/optimize/_simulated_annealing.py

```diff
@@ -0,0 +1,559 @@
+
+        # Use the new point only if it is valid and return a better results
+        if is_valid and mres.fun < fun_temp:
+            return mres.fun, mres.x
+        else:
+            return fun_temp, x_tmp 
+
+
+def simulated_annealing(func, x0, bounds, maxiter=1000, local_search_options={},
+        initial_temp=5230., visit=2.62, accept=-5.0, maxfun=1e7, seed=None,
+        no_local_search=False):
+    """
+    Find the global minimum of a function using the Simulated Dual Annealing
+    algorithm
+
+    Parameters
+    ----------
+    func : callable
+        The objective function to be minimized.  Must be in the form
+        ``f(x, *args)``, where ``x`` is the argument in the form of a 1-D array
+        and ``args`` is a  tuple of any additional fixed parameters needed to
+        completely specify the function.
+    x0 : ndarray
+        The starting coordinates. If ``None`` is provided, initial
+        coordinates are automatically generated.
+    bounds : sequence
+        Bounds for variables.  ``(min, max)`` pairs for each element in ``x``,
+        defining the lower and upper bounds for the optimizing argument of
+        `func`. It is required to have ``len(bounds) == len(x)``.
+        ``len(bounds)`` is used to determine the number of parameters in ``x``.
+    maxiter : int, optional
+        The maximum number of simulated_annealing  iterations. Increase this
+        value if the objective function is very complicated with high
+        dimensions.
+    local_search_options : dict, optional
+        Extra keyword arguments to be passed to the local minimizer
+            ``scipy.optimize.minimize()`` Some important options could be:
+            method : str
+                The minimization method (e.g. ``"L-BFGS-B"``)
+            args : tuple
+                Extra arguments passed to the objective function (``func``) and
+                its derivatives (Jacobian, Hessian).
+    initial_temp : float, optional
+        The initial temperature, use higher values to facilitates a wider
+        search of the energy landscape, allowing simulated_annealing  to escape
+        local minima that it is trapped in.
+    visit : float, optional
+        Parameter for visiting distribution. Higher values give the visiting
+        distribution a heavier tail, this makes the algorithm jump to a more
+        distant region. The value range is (0, 3]
+    accept : float, optional
+        Parameter for acceptance distribution. It is used to control the
+        probability of acceptance. The lower the acceptance parameter, the
+        smaller the probability of acceptance. It has to be any negative value.
+    maxfun : int, optional
+        Soft limit for the number of objective function calls. If the
+        algorithm is in the middle of a local search, this number will be
+        exceeded, the algorithm will stop just after the local search is
+        done.
+    seed : int or `np.random.RandomState`, optional
```

Comment:
I prefer:
`` seed : {int, `np.random.RandomState`}, optional``

Reply:
Could not the user just set:
```
np.random.seed()
```
in order to get something reproducible?

Do we actually need a parameter for that?

Reply:
Thanks. Doc adjusted. I think we do need a parameter for that and using `RandomState` was a better option, this was discussed with @js850 and @andyfaff .